app/scripts/views/NodeViews/List.js

```diff
@@ -0,0 +1,109 @@
+define(['backbone', 'underscore', 'jquery', 'BaseNodeView'], function (Backbone, _, $, BaseNodeView) {
+
+    return BaseNodeView.extend({
+
+        initialize: function (args) {
+            BaseNodeView.prototype.initialize.apply(this, arguments);
+        },
+
+        innerTemplate: _.template($('#node-list-template').html()),
+
+        getCustomContents: function () {
+            var json = this.model.toJSON();
+
+            return this.innerTemplate(json);
+        },
+
+        updatePorts: function () {
+            var type = this.model.get('type');
+            var ex = this.model.get('extra');
+            var exCopy = JSON.parse(JSON.stringify(ex));
+            var ignoreDefaults = type.inputs.map(function (x) {
+                return !x.useDefault;
+            });
+            exCopy.varInputs = type.inputs.map(function (port) {
+                return { name: port.name, type: port.typeName };
+            });
+
+            this.model.set('ignoreDefaults', ignoreDefaults);
+
+            this.model.workspace.setNodeProperty({ property: "extra", _id: this.model.get('_id'), newValue: exCopy, oldValue: ex });
+
+            this.render();
+            this.model.trigger('updateRunner');
+            this.model.workspace.run();
+        },
+
+        renderNode: function () {
+
```

Comment:
Ensure this doesn't get called multiple times.  NodeView does not give this guarantee:

https://github.com/pboyer/flood/blob/master/app/scripts/views/NodeViews/Base.js